src/tpm2_wrap.c

```diff
@@ -172,6 +172,249 @@ int wolfTPM2_Init(WOLFTPM2_DEV* dev, TPM2HalIoCb ioCb, void* userCtx)
     return rc;
 }
 
+#ifdef WOLFTPM2_ENABLE_CSHARP
+WOLFTPM2_DEV *wolfTPM2_SimpleInit(void)
+{
+    WOLFTPM2_DEV *dev = NULL;
+
+    dev = (WOLFTPM2_DEV *) XMALLOC(sizeof(WOLFTPM2_DEV), NULL,
+                                   DYNAMIC_TYPE_TMP_BUFFER);
+    if (dev == NULL) {
+        return NULL;
+    }
+
+    if (wolfTPM2_Init(dev, NULL, NULL) != TPM_RC_SUCCESS) {
+        return NULL;
+    }
+
+    return dev;
+}
+
+int wolfTPM2_SimpleCleanup(WOLFTPM2_DEV *dev)
+{
+    wolfTPM2_Cleanup(dev);
+    XFREE(dev, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM2_KEYBLOB* wolfTPM2_GetEmptyKeyBlob(void)
+{
+    WOLFTPM2_KEYBLOB* blob = NULL;
+
+    blob = (WOLFTPM2_KEYBLOB *) XMALLOC(sizeof(WOLFTPM2_KEYBLOB), NULL,
+                                      DYNAMIC_TYPE_TMP_BUFFER);
+    if (blob == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(blob, 0, sizeof(WOLFTPM2_KEYBLOB));
+    return blob;
+}
+
+int wolfTPM2_SimpleOpenExisting(WOLFTPM2_DEV* dev)
```

Comment:
I see no value is adding this API. You should be able to call `wolfTPM2_OpenExisting` directly.

Reply:
I don't want them to be able to set ioCb and userCtx to anything but NULL. 
This also reduces the potential for questions like..."Oh...what is the callback for?"

Reply:
Then please rename. I don't like "simple". How about `wolfTPM2_OpenExistingDev`?

Reply:
No problem.  That I can do.